### ISSUE TITLE
Media players Whirligig, Skybox can no longer access files in home folder

### ISSUE THREAD

**Issue by yaomtc** [ValveSoftware/steam-runtime#549]
#### Your system information

* Steam Runtime Version: 
* Distribution (e.g. Ubuntu 18.04): Arch Linux
* Link to your full system information (Help -> System Information) in a [Gist](https://gist.github.com/): 
https://gist.github.com/yaomtc/6bfbff64fcc5ddb7a2add9c27d6bfb21
* Have you checked for system updates?: Yes
* What compatibility tool are you using?: Proton 5.13+
* If you are using Steam Linux Runtime, or Proton 5.13 or newer: What versions are listed in `SteamLinuxRuntime_soldier/VERSIONS.txt`? 
```
depot	0.20221018.74			# Overall version number
pressure-vessel	0.20221014.0			
scripts	v0.20220823.0-0-gcc4e44f			# Entry point scripts, etc.
soldier	0.20221017.1	soldier	0.20221017.1	# soldier_platform_0.20221017.1/
```

#### Please describe your issue in as much detail as possible:
In the past those players were able to access folders I wanted by using this launch option:

`STEAM_COMPAT_MOUNTS=/home/myusername/asdf:/mnt/WD1TBlue/ %command%`

(Or, /home/myusername instead of /home/myusername/asdf)

This no longer works. They can no longer see inside that directory inside the home folder. (Whirligig just refuses to do/see anything with the directory, while Skybox crashes when I try to access it.) They can access the other one just fine, though.

[slr-app451650-t20221208T134006.log](https://github.com/ValveSoftware/steam-runtime/files/10189005/slr-app451650-t20221208T134006.log)
[slr-app721090-t20221208T134054.log](https://github.com/ValveSoftware/steam-runtime/files/10189006/slr-app721090-t20221208T134054.log)
[steam-451650.log](https://github.com/ValveSoftware/steam-runtime/files/10189018/steam-451650.log)
[steam-546560.txt](https://github.com/ValveSoftware/steam-runtime/files/10189019/steam-546560.txt)
[steam-721090.log](https://github.com/ValveSoftware/steam-runtime/files/10189020/steam-721090.log)

Not sure where the Proton log goes from 

#### Steps for reproducing this issue:

1. Launch SteamVR 
2. Launch Skybox VR Player or Whirligig Media Player in SteamVR with STEAM_COMPAT_MOUNTS in launch options
3. Attempt to access the specified folders from in the software

**Comment by smcv** [comment]
From the log, it seems that the container runtime framework is at least *trying* to set these up:

```
13:40:07.170299: pressure-vessel-wrap[33508]: I: Bind-mounting STEAM_COMPAT_MOUNTS="...:/home/myusername/asdf:..." from the current env as STEAM_COMPAT_MOUNTS="...:/home/myusername/asdf:..." in the host
13:40:07.170343: pressure-vessel-wrap[33508]: I: Bind-mounting STEAM_COMPAT_MOUNTS="...:/mnt/WD1TBlue/:..." from the current env as STEAM_COMPAT_MOUNTS="...:/mnt/WD1TBlue:..." in the host
```

Please try with `STEAM_LINUX_RUNTIME_VERBOSE=1` added to the environment? You should get a considerably larger log file.

**Comment by yaomtc** [comment]
`STEAM_LINUX_RUNTIME_VERBOSE=1 STEAM_LINUX_RUNTIME_LOG=1 PROTON_LOG=1 steam`

Like this, or is it redundant?

**Comment by smcv** [comment]
Note that if your home directory is `/home/myusername`, then all Steam games and applications are meant to get access to `/home/myusername` by default, whether you put it in `STEAM_COMPAT_MOUNTS` or not - unless you specifically ask the container runtime framework to use a special per-game home directory (an experimental feature), but there's certainly no reason to do that if you *want* the games/apps to have access to your home directory.

Is there anything special about your home directory: symbolic links, unusual filesystems (anything other than ext2/ext3/ext4/btrfs/xfs), owned by a different user ID, special ACLs, that sort of thing?

> Whirligig just refuses to do/see anything with the directory

What does "just refuses" mean, specifically?

> Skybox crashes when I try to access it

That suggests that the directory is shared with the application, but something is unusual about it in a way that triggers a crash inside Proton/Wine.

> They can access the other one just fine

What is "the other one"? Do you mean these applications can access `/mnt/WD1TBlue` successfully, but accessing `/home/myusername` fails?

**Comment by smcv** [comment]
> `STEAM_LINUX_RUNTIME_VERBOSE=1 STEAM_LINUX_RUNTIME_LOG=1 PROTON_LOG=1 steam`
> 
> Like this?

Yes, like that.

**Comment by yaomtc** [comment]
> What does "just refuses" mean, specifically?

By choosing Z: I can navigate to /home/, but then selecting /myusername/ doesn't do anything, I'm still in /home/.

> Do you mean these applications can access /mnt/WD1TBlue successfully, but accessing /home/myusername fails?

Correct, that other drive works just fine with the players but my home folder doesn't.

My home folder isn't on a separate partition or anything. I wonder if it could be a permissions issue?


[slr-app451650-t20221208T142236.log](https://github.com/ValveSoftware/steam-runtime/files/10189261/slr-app451650-t20221208T142236.log)
[steam-451650.log](https://github.com/ValveSoftware/steam-runtime/files/10189268/steam-451650.log)

**Comment by yaomtc** [comment]
I found some newlines in a few filenames in the folder. (Not sure how I managed to get a newline in a filename.) Removed those and fixed some other filenames that had possibly bad characters. Whirligig is now able to access /home/username/asdf too, though not via /home/username/, only via the "X:" drive shortcut. Skybox wasn't show the drive letter mappings at first, but they showed up after an uninstall and reinstall.
